trio/_core/_io_windows.py

```diff
@@ -368,7 +368,7 @@ def _get_base_socket(sock: _HasFileNo | int | Handle) -> Handle:
                     "https://github.com/python-trio/trio/issues/new, "
                     "and include the output of running: "
                     "netsh winsock show catalog"
-                )
+                ) from ex
```

Comment:
this is a functionality change, though I have no clue if it's bad.

Reply:
My logic changing this was that maybe seeing the full OS error in the traceback might be helpful, but up for debate.

Reply:
Can probably keep it in, but add a note in newsfragments?

Reply:
Added not in f80b22a5